### PR TITLE
Feature/dc 547 task app module

### PR DESCRIPTION
**Pull request by oleksii-symon-corva-ai**
[DC-547](https://corvaqa.atlassian.net/browse/DC-547)

- added TaskApp with tests
- moved all models to `corva/models/` dir
- moved `corva/event/loader/` dir to `corva/loader/`
- `Event` changes:
-- deleted not needed init together with obsolete `data` keyword param
-- deleted useless tests, that were testing UserList
-- moved `Event` class to `corva/event.py`, which caused many import modifications
- renamed `contants.py` into `types.py`

### CODE REVIEW

File: tests/app/test_task.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+
+def test_pre_process_calls_base(mocker: MockerFixture, task_app, task_context_factory):
+    context = task_context_factory()
+
+    super_pre_process_mock = mocker.patch.object(BaseApp, 'pre_process')
+
+    task_app.pre_process(context=context)
+
+    super_pre_process_mock.assert_called_once_with(context=context)
```
[the remark this answers is not in the document]
Comment (oleksii-symon-corva-ai) [reply tests/app/test_task.py:44]: got rid of such calls

File: corva/app/task.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            status='success',
+            data=UpdateTaskInfoData(payload=context.save_data)
+        )
+
+    def on_fail(self, context: TaskContext, exception: Exception) -> None:
+        super().on_fail(context=context, exception=exception)
+
+        data = UpdateTaskInfoData(
+            fail_reason=str(exception),
+            payload={'error': ''.join(traceback.TracebackException.from_exception(exception).format())}
```
[the remark this answers is not in the document]
Comment (oleksii-symon-corva-ai) [reply corva/app/task.py:42]: I've left only `fail_reason` with empty payload

File: corva/app/utils/context.py [hunk 3]
```diff
@@ -17,3 +18,8 @@ class ScheduledContext(BaseContext):
 
 class StreamContext(BaseContext):
     state: RedisState
+
+
+class TaskContext(BaseContext):
+    task: TaskData
+    save_data: dict = {}
```
[the remark this answers is not in the document]
Comment (oleksii-symon-corva-ai) [reply corva/app/utils/context.py:25]: Renamed into `task_result`. Data that should be saved after task execution.

File: tests/conftest.py [hunk 4]
```diff
@@ -7,17 +7,20 @@
...
 from corva.event.data.stream import Record, StreamEventData
+from corva.event.data.task import TaskEventData
 from corva.event.event import Event
 from corva.state.redis_adapter import RedisAdapter
 from corva.state.redis_state import RedisState
 
 APP_KEY = 'provider.app-name'
 SCHEDULED_EVENT_FILE_PATH = 'data/tests/scheduled_event.json'
 STREAM_EVENT_FILE_PATH = 'data/tests/stream_event.json'
+TASK_EVENT_FILE_PATH = 'data/tests/task_event.json'
```
[the remark this answers is not in the document]
Comment (oleksii-symon-corva-ai) [reply tests/conftest.py]: moved to `tests/test_data/` dir

**Review by ville — APPROVED** [review]

File: tests/test_data/task_event.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+  "task_id": "dc5c16e4-c0e3-43fc-85b2-83b08b9da7d0",
+  "version": 2
+}
```
Comment (finomayato) [inline tests/test_data/task_event.json:4]: can we move it to the code, to now create redundant files?

File: corva/models/scheduled.py [hunk 6]
```diff
@@ -1,6 +1,12 @@
...
 from datetime import datetime
-from corva.event.data.base import BaseEventData
+from typing import Optional
+
+from corva.models.base import BaseContext, BaseEventData
+from corva.state.redis_state import RedisState
+
+
+class ScheduledContext(BaseContext):
+    state: RedisState
```
Comment (finomayato) [inline corva/models/scheduled.py:9]: do we really want to expose internal implementation to the user?

**Review by finomayato — APPROVED** [review]
Much better now

File: corva/models/scheduled.py (hunk 6 above)
Comment (oleksii-symon-corva-ai) [reply corva/models/scheduled.py:9]: will take a note to discuss this

**Pull request merged**